src/v0/destinations/bqstream/transform.js

```diff
@@ -102,38 +103,52 @@ const processRouterDest = (inputs) => {
   if (errorRespEvents.length > 0) {
     return errorRespEvents;
   }
-
-  const eventsChunk = []; // temporary variable to divide payload into chunks
-  const errorRespList = [];
-
-  inputs.forEach((event) => {
-    try {
-      if (event.message.statusCode) {
-        // already transformed event
-        eventsChunk.push(event);
-      } else {
-        // if not transformed
-        let response = process(event);
-        response = Array.isArray(response) ? response : [response];
-        response.forEach((res) => {
-          eventsChunk.push({
-            message: res,
-            metadata: event.metadata,
-            destination: event.destination,
-          });
-        });
+  const groupedEvents = getGroupedEvents(inputs);
+  const finalResp = [];
+  groupedEvents.forEach((eventList) => {
+    let eventsChunk = []; // temporary variable to divide payload into chunks
+    let errorRespList = [];
+    if (eventList.length > 0) {
+      eventList.forEach((event) => {
+        try {
+          if (event.message.statusCode) {
+            // already transformed event
+            eventsChunk.push(event);
+          } else {
+            // if not transformed
+            let response = process(event);
+            response = Array.isArray(response) ? response : [response];
+            response.forEach((res) => {
+              eventsChunk.push({
+                message: res,
+                metadata: event.metadata,
+                destination: event.destination,
+              });
+            });
+          }
+        } catch (error) {
+          const errRespEvent = handleRtTfSingleEventError(event, error, DESTINATION);
+          // divide the successful payloads till now into batches
```

Comment:
why do we need to batch this way? is it not possible process events all and then handling batching?

Reply:
Yes! As you said, we were handling it just like this previously. 

example : 
u1 is sending -->  [ track1, track2, identify1, track3, track 4, identify 4, identify 5, track 5, track 6 ], now, the transformer was sending [[ track1, track2, track3, track 4, track 5  ], [identify1],  [identify 4], [ identify 5 ]]. 

In the case of the bq stream, the identify event is not supported.  But, to maintain the event order and user journey, we need to break the batches anyway. For example,

u1 is sending -->  [ track1, track2, identify1, track3, track 4, identify 4, identify 5, track 5 ], now, to maintain the above scenarios, the transformer is sending [[ track1, track2 ], [identify1], [track3, track 4], [identify 4], [ identify 5 ], [ track 6]]. 

With little change, we could do it this way. Do you recommend keeping the catch part only for error-catching and writing an event order handling util function to handle this?

Reply:
code is hard to follow now. what ever may be requirement, code should be easy to understand

Reply:
addressed